UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs

```diff
@@ -990,4 +983,10 @@ public void CmdServerReplaceItemInInventory(GameObject gameObjectSent, uint id,
 			Inventory.ServerTransfer(gameObjectSent.PickupableOrNull().ItemSlot, slot, ReplacementStrategy.DropOther);
 		}
 	}
+
+	[Command]
+	public void CmdDoEmote(GameObject player, string emoteName)
```

Comment:
get the GameObject player,  
 As this.game object because a hacked client could send any player  to do emote

Reply:
done